### DISCUSSION TITLE
physical wiring

### DISCUSSION THREAD

**Discussion by Paradisebaker** [freakydude/klipper-serial-btt#1] [Q&A] [answered]
Hi Freakydude
I like your idea and would like to try it out, but I am unclear about the physical connections.
My current setup is like this:
Artillery SW X1 with stock TFT28 and stock MKS GenL.
Odroid XU4 as SBC with Klipper/Mainsail, latest versions.
special: not using regular USB cable, but have a USB2TTL adapter going directly to the D1 and D2 pins of AUX1 (where the TFT used to be). TFT is connected to Serial2 (in marlin speak) using D16 and D17 for TX/RX and the associated GND and 5V, all in EXT1.
all connections are using 250000 baud. the tft of course shows "no printer connected"

how do I need to adjust your files to reflect these connections. or do I need to connect the TFT directly to the UART of the SBC ?
(if you explain this in RPi terms, I can "translate" it to the XU4)
It would be great if this could be done, I would hate to waste the TFT.

One other question, if your macros conflict with mainsail, would it be better to just use Fluidd ? or is your macro compensation taking care of all conflicts?

Thanks so much for doing this and helping me out.
best regards
markus

**Comment by freakydude** [comment]
Hi Markus,

nice that you are interested in my project - and already found it. I had unfortunately not yet time and mood to describe the whole project - especially the wiring.

I suggest, you give me a few days time to create the blog post. Then try it out and ask questions. So there is a chance test and improve it in your constellation. 

For clarification: The macros don't conflict with mainsail itself, perhaps I didn't describe it well. I made them especially for the BTT-TFT and Mainsail in combination.

There is an "official" additional mainsail macro repository [mainsail-config](https://github.com/mainsail-crew/mainsail-config). If you are fully integrating that in your installation - it will conflict.

I didn't work with Fluidd, thought the two projects are basically the same. Also a point for improvements and feedback.


When you can't wait, a small hint how I did it. I connected it to the first UART of the 40GPIO pins from the raspberry. Caution: You will also need a level shifter 3/5V that you don't "burn" your raspberry pi. 

Meanwhile, if you got something, please let me know, 

With best regards,
freakyDude

**Comment by Paradisebaker** [comment]
servus.Ich nehme mal an, wir können in deutsch schreiben. das mit der Verkabelung kann warten, ich lerne gerade erst mal laufen mit klipper, der umstieg von marlin ist nicht so leicht. Wegen level shifter, da der XU4 sogar nur 1,8v verträgt, ist das noch wichtiger. könnte man da nicht auch einen USB2TTL nehmen? am tft wäre der anschluss ja ident, auf SBC seite dann eben usb.lgMarkus Sent from my mobile phone, please excuse brevity and/or spelling errors 
-------- Ursprüngliche Nachricht --------Von: freakyDude ***@***.***> Datum: 13.04.23  20:27  (GMT+01:00) An: freakydude/klipper-serial-btt ***@***.***> Cc: Paradisebaker ***@***.***>, Author ***@***.***> Betreff: Re: [freakydude/klipper-serial-btt] physical wiring (Discussion #1) 
Hi Markus,
nice that you are interested in my project - and already found it. I had unfortunately not yet time and mood to describe the whole project - especially the wiring.
I suggest, you give me a few days time to create the blog post. Then try it out and ask questions. So there is a chance test and improve it in your constellation.
For clarification: The macros don't conflict with mainsail itself, perhaps I didn't describe it well. I made them especially for the BTT-TFT and Mainsail in combination.
There is an "official" additional mainsail macro repository mainsail-config. If you are fully integrating that in your installation - it will conflict.
I didn't work with Fluidd, thought the two projects are basically the same. Also a point for improvements and feedback.
When you can't wait, a small hint how I did it. I connected it to the first UART of the 40GPIO pins from the raspberry. Caution: You will also need a level shifter 3/5V that you don't "burn" your raspberry pi.
Meanwhile, if you got something, please let me know,
With best regards,
freakyDude

**Reply by freakydude** [reply]
Ich gehe davon aus, dass das auch funktionieren dürfte. Brauchst halt am Ende eine serielle Schnittstelle, dort wo auch Klipper läuft. Und das solltest du ja mit USB2ttl haben +Level shifter

**Comment by Paradisebaker** [comment]
naja, das wär ja grad der Vorteil,  level shifter brauchst dann keinen mehr, usb hat ja 5v out.Sent from my mobile phone, please excuse brevity and/or spelling errors 
-------- Ursprüngliche Nachricht --------Von: freakyDude ***@***.***> Datum: 13.04.23  22:39  (GMT+01:00) An: freakydude/klipper-serial-btt ***@***.***> Cc: Paradisebaker ***@***.***>, Author ***@***.***> Betreff: Re: [freakydude/klipper-serial-btt] physical wiring (Discussion #1) 
Ich gehe davon aus, dass das auch funktionieren dürfte. Brauchst halt am Ende eine serielle Schnittstelle, dort wo auch Klipper läuft. Und das solltest du ja mit USB2ttl haben +Level shifter

**Comment by Paradisebaker** [comment, marked as answer]
switching back to english so more people can get the info about this great project.
I got it working now with a USB2TTL adapter with the GND, TX and RX connected to the displays UART ports, 5v comes from the printer board. This way, if I power toggle the printer from Klipper (well, actually from Moonraker), the screen only lights up if the board is powered.

**Reply by freakydude** [reply]
Got the blog post not yet online 😱
But really nice to hear. 

Congrat, you are probably the second freaky guy on the world doing that. And nice idea to get the power from board not from the pi. 

Now you need to push something useful an it's manifested for years

😉👍🏼

**Reply by freakydude** [reply]
My blog post is [online](https://blog.freakydu.de/posts/2023-03-23-use_bigtreetech_touchscreenfirmware_with_klipper_on_sidewinder_x2/)
